### ISSUE TITLE
When importing a file as it is done in python 3.8, in python it works correctly but in pytest it doesn't

### ISSUE THREAD

**Issue by LuckJMG** [pytest-dev/pytest#7502]
## Introduction

Hi! I have been doing a small project in which everything worked well, but when I came back today it no longer worked for me taking pytest, now I explain it to you.

---

## Description

First, this is a simplified version of the structure of my project

```
src
└───mypkg
    ├───__init__.py
    └───core.py

└───tests
    ├───__init__.py
    └───test_core.py
```

In mypkg/__init__.py
`
from core import Addition
`

In mypkg/core.py
```
class Addition():

    def addition(self, x, y):
        z = x +y
        return z
```

There is nothin in tests/__init__.py 

And in tests/test_core.py
```
from mypkg import Addition

class TestCore(Addition):

    def test_addition(self):
        assert self.addition(1, 2) == 3
```

So when I run mypkg/__ init__.py (to check if files are imported well) in python, it runs smoothly, but when I run pytest in tests/test_core.py this happens:
```
========================================================= test session starts ========================================================== 
platform win32 -- Python 3.8.4, pytest-6.0.0rc1, py-1.9.0, pluggy-0.13.1
rootdir: C:\Users\Luck\Desktop\pytest test
collected 0 items / 1 error                                                                                                              

================================================================ ERRORS ================================================================ 
_________________________________________________ ERROR collecting tests/test_core.py __________________________________________________ 
ImportError while importing test module 'C:\Users\Luck\Desktop\pytest test\tests\test_core.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
C:\Program Files\Python38\lib\importlib\__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests\test_core.py:1: in <module>
    from mypkg import Addition
mypkg\__init__.py:1: in <module>
    from core import Addition
E   ImportError: cannot import name 'Addition' from 'core' (C:\Users\Luck\Projects\Scripts\RealFractions\real_fractions\core.py)
======================================================= short test summary info ======================================================== 
ERROR tests/test_core.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! 
=========================================================== 1 error in 0.30s ===========================================================
```

And what strikes me is that when in mypkg/__ init__.py I change it like this:
```
from .core import Addition
```

In this case when I put a dot (.) To the file, in pytest everything runs normally:
```
========================================================= test session starts ========================================================== 
platform win32 -- Python 3.8.4, pytest-6.0.0rc1, py-1.9.0, pluggy-0.13.1
rootdir: C:\Users\Luck\Desktop\pytest test
collected 1 item                                                                                                                         

tests\test_core.py .                                                                                                              [100%] 

========================================================== 1 passed in 0.25s ===========================================================
```
but when i run mypkg/__ init__.py in python it happens this is:
```
Traceback (most recent call last):
  File ".\mypkg\__init__.py", line 1, in <module>
    from .core import Addition
ImportError: attempted relative import with no known parent package
```

I have searched everywhere how to solve the problem, but I did not find a solution, so I wanted to see if they could help me.
Thank you for your attention and have a good day

---

## Output of `pip list`
```
Package                       Version
----------------------------- ----------
alabaster                     0.7.12
appdirs                       1.4.4
astroid                       2.4.1
atomicwrites                  1.4.0
attrs                         19.3.0
Babel                         2.8.0
black                         19.10b0
bleach                        3.1.5
certifi                       2020.4.5.1
chardet                       3.0.4
click                         7.1.2
colorama                      0.4.3
docutils                      0.16
hunter                        3.1.3
idna                          2.9
imagesize                     1.2.0
iniconfig                     1.0.0
isort                         4.3.21
Jinja2                        2.11.2
keyring                       21.2.1
lazy-object-proxy             1.4.3
MarkupSafe                    1.1.1
mccabe                        0.6.1
more-itertools                8.3.0
packaging                     20.4
pathspec                      0.8.0
pip                           20.1.1
pkginfo                       1.5.0.1
pluggy                        0.13.1
py                            1.9.0
Pygments                      2.6.1
pylint                        2.5.2
pyparsing                     2.4.7
pytest                        6.0.0rc1
pytz                          2020.1
pywin32-ctypes                0.2.0
readme-renderer               26.0
regex                         2020.5.14
requests                      2.23.0
requests-toolbelt             0.9.1
rstcheck                      3.3.1
setuptools                    47.1.0
six                           1.15.0
snowballstemmer               2.0.0
Sphinx                        3.0.4
sphinxcontrib-applehelp       1.0.2
sphinxcontrib-devhelp         1.0.2
sphinxcontrib-htmlhelp        1.0.3
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.3
sphinxcontrib-serializinghtml 1.1.4
toml                          0.10.1
tqdm                          4.46.0
twine                         3.1.1
typed-ast                     1.4.1
urllib3                       1.25.9
wcwidth                       0.2.2
webencodings                  0.5.1
wheel                         0.34.2
wrapt                         1.12.1
```

---

## Versions and OS
- python => 3.8.4
- pytest => 6.0.0rc1
- OS => Windows 10

**Comment by asottile** [comment]
running a file inside a module is **not** equivalent to how that module will act when used as a library

you'll want to use `python -m mypkg` and not `python mypkg/__init__.py` -- the latter assumes `__init__.py` is a script and will execute it (ignoring your package layout, and [adding `mypkg` at the beginning of sys.path](https://github.com/asottile/scratch/wiki/PythonPathSadness))

the second way, with the explicit relative import is correct (and pytest is correct to show that the implicit relative import is incorrect)

tl;dr, always use `python -m` when executing modules

**Comment by LuckJMG** [comment]
Aaaaah, thank you very much then you must run the module like this `python -m mypkg .__ init__`.
Thanks for your attention and sorry for the inconvenience, I am still learning python.
Have a nice day.
